src/correlations.jl

```diff
@@ -1,246 +1,186 @@
-export SpectrumSolver, FFTCorrelation, ExponentialSeries
-export correlation_3op_2t, correlation_2op_2t, correlation_2op_1t, spectrum
-
-abstract type SpectrumSolver end
-
-struct FFTCorrelation <: SpectrumSolver end
-
-struct ExponentialSeries{T<:Real,CALC_SS} <: SpectrumSolver
-    tol::T
-    ExponentialSeries(tol::T, calc_steadystate::Bool = false) where {T} = new{T,calc_steadystate}(tol)
+export correlation_3op_2t, correlation_3op_1t, correlation_2op_2t, correlation_2op_1t
+
+function _check_correlation_time_list(tlist::AbstractVector)
+    any(t -> t == 0, tlist) ||
+        throw(ArgumentError("The time list for calculating correlation function must contain the element `0`"))
+    all(>=(0), tlist) ||
+        throw(ArgumentError("All the elements in the time list for calculating correlation function must be positive."))
+    return nothing
 end
 
-ExponentialSeries(; tol = 1e-14, calc_steadystate = false) = ExponentialSeries(tol, calc_steadystate)
-
 @doc raw"""
-    correlation_3op_2t(H::QuantumObject,
-        ψ0::QuantumObject,
-        t_l::AbstractVector,
-        τ_l::AbstractVector,
+    correlation_3op_2t(H::AbstractQuantumObject,
+        ψ0::Union{Nothing,QuantumObject},
+        tlist::AbstractVector,
+        τlist::AbstractVector,
+        c_ops::Union{Nothing,AbstractVector,Tuple},
         A::QuantumObject,
         B::QuantumObject,
-        C::QuantumObject,
-        c_ops::Union{Nothing,AbstractVector,Tuple}=nothing;
+        C::QuantumObject;
         kwargs...)
 
-Returns the two-times correlation function of three operators ``\hat{A}``, ``\hat{B}`` and ``\hat{C}``: ``\left\langle \hat{A}(t) \hat{B}(t + \tau) \hat{C}(t) \right\rangle``
+Returns the two-times correlation function of three operators ``\hat{A}``, ``\hat{B}`` and ``\hat{C}``: ``\left\langle \hat{A}(t) \hat{B}(t + \tau) \hat{C}(t) \right\rangle`` for a given initial state ``|\psi_0\rangle``.
 
-for a given initial state ``\ket{\psi_0}``.
+If the initial state `ψ0` is given as `nothing`, then the [`steadystate`](@ref) will be used as the initial state. Note that this is only implemented if `H` is constant ([`QuantumObject`](@ref)).
 """
 function correlation_3op_2t(
-    H::QuantumObject{<:AbstractArray{T1},HOpType},
-    ψ0::QuantumObject{<:AbstractArray{T2},StateOpType},
-    t_l::AbstractVector,
-    τ_l::AbstractVector,
-    A::QuantumObject{<:AbstractArray{T3},OperatorQuantumObject},
-    B::QuantumObject{<:AbstractArray{T4},OperatorQuantumObject},
-    C::QuantumObject{<:AbstractArray{T5},OperatorQuantumObject},
-    c_ops::Union{Nothing,AbstractVector,Tuple} = nothing;
+    H::AbstractQuantumObject{DataType,HOpType},
+    ψ0::Union{Nothing,QuantumObject{<:AbstractArray{T1},StateOpType}},
+    tlist::AbstractVector,
+    τlist::AbstractVector,
+    c_ops::Union{Nothing,AbstractVector,Tuple},
```

Comment:
Also here and in all the other cases below.

Reply:
just specify `nothing` or empty `c_ops` list.

QuTiP can only specify empty `c_ops` for this situation.